### ISSUE TITLE
[BUG] converting sqlite to postgresdb

### ISSUE THREAD

**Issue by eisvogelETH** [linuxserver/docker-nextcloud#441]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

![image](https://github.com/linuxserver/docker-nextcloud/assets/19222231/ea9f5ca3-ae6a-4080-b431-ebcc179c79e0)
Unfortunately I am not able to convert the sqlite to a postgres db with the comment below...

### Expected Behavior

working migration

### Steps To Reproduce

try to migrate linuxserver/nextcloud (tried to upgrade to the tag latest 4h ago (29.0.2) and still not working

### Environment

```markdown
- OS: ubuntu 22.04 lts
- How docker service was installed:
```

### CPU architecture

x86-64

### Docker creation

```bash
nextcloud:
    image: lscr.io/linuxserver/nextcloud:latest
    container_name: nextcloud
    environment:
      - PUID={{ puid }}
      - PGID={{ pgid }}
      - TZ={{ timezone }}
    volumes:
      - "{{ docker_dir }}/nextcloud/appdata:/config" 
      - "{{ docker_dir }}/nextcloud/data:/data"
    restart: unless-stopped
    depends_on:
      - nextcloud-db
    links:
      - nextcloud-db
    ports:
      - '4443:443'
  
  nextcloud-db:
    image: postgres:15
    restart: always
    container_name: nextcloud_db
    environment:
      - POSTGRES_USER={{ nextcloud_user }}
      - POSTGRES_PASSWORD={{ nextcloud_pw }}
      - POSTGRES_DB={{ nextcloud_db }}
    volumes:
      - "{{ docker_dir }}/nextcloud/db:/var/lib/postgresql/data"
    healthcheck:
      test: ["CMD-SHELL", "sh -c 'pg_isready -U {{ nextcloud_user }} -d nextcloud'"]
      interval: 10s
      timeout: 3s
      retries: 3
```

### Container logs

```bash
docker exec -it nextcloud bash
root@xxxx:/# occ db:convert-type --port 5432  --all-apps  --password=XXXXX pgsql user host db

In ConvertType.php line 183:
                                                                              
  This command is temporarily disabled (until the next maintenance release).  
                                                                              

db:convert-type [--port PORT] [--password PASSWORD] [--clear-schema] [--all-apps] [--chunk-size CHUNK-SIZE] [--] <type> <username> <hostname> <database>
```

**Comment by j0nnymoe** [comment]
That seems more like something nextcloud disabled rather than anything to do with us.

**Comment by Investigamer** [comment]
Is there a nextcloud version you can safely revert to in order to convert your database, then jump back to latest? I also picked a terrible time to finally switch to postgres unfortunately.

**Comment by joshtrichards** [comment]
@Investigamer Anything <29. But you can't downgrade so it would have to be using a database snapshot from prior to upgrading to v29.

Context: This command was disabled very recently (in 29.0.1).  So alternatively, If you can wait another maintenance release cycle or two (maintenance releases are published at least monthly), things should be back to normal and you'd be able to do the conversion without making other changes or reverting to an old backup.

Whatever you do, don't try doing it with 29.0.0. This command was disabled intentionally for the 29.0.1 release to prevent people inadvertently stumbling onto a data loss regression in this command that arose from some other v29 specific db change work.

**Comment by johnny2678** [comment]
wow, bad timing.

Need to save off my `occ` command so I don't have to spend 15 minutes constructing it next time 😂

**Comment by eisvogelETH** [comment]
thank you for the replies guys! I guess i can close the issue now that we have a solution to this?

**Comment by Investigamer** [comment]
Thanks for the responses fellas. Just saw a new Nextcloud maintenance version has gone out, does anyone know if this was fixed in 29.0.3? Scouring the changelog, see some mention of changes with ocp but nothing specific. Also having trouble finding exactly where in nextcloud's myriad of repositories the progress if this issue is tracked.

**Comment by CorneliousJD** [comment]
Still not resolved in 29.0.4 sadly...

**Comment by johnny2678** [comment]
> Still not resolved in 29.0.4 sadly...

Thanks for letting us know

**Comment by nate-moo** [comment]
Based on this PR, it looks like its being slated for v30 
https://github.com/nextcloud/server/pull/46931

**Comment by joshtrichards** [comment]
It'll be backported to all applicable non-EoL majors once merged; it's a bug fix. i.e. the nearest maintenance release for v29 will get it too.

**Comment by Investigamer** [comment]
For anyone wondering if you can backup your instance and spin up an instance of v30.0.0-beta4, no the command is still disabled as of beta 4. Things are happening at least.

**Comment by joshtrichards** [comment]
If you're in a position to test, see here:

https://github.com/nextcloud/server/issues/45257#issuecomment-2276730407

**Comment by Investigamer** [comment]
Unfortunately I'm not in a position to test, but just to update everyone this likely won't be fixed till at least 29.0.7 based on current milestones.

**Comment by Meowcat285** [comment]
> Unfortunately I'm not in a position to test, but just to update everyone this likely won't be fixed till at least 29.0.7 based on current milestones.

Sadly still not fixed on 29.0.7

**Comment by Investigamer** [comment]
> Sadly still not fixed on 29.0.7

This issue has now been marked with milestone 31 on the issue tracker... welp fellas, I think at this point I'm just going to build a fresh nextcloud installation 😅

**Comment by drizuid** [comment]
im going to go ahead and close this as it is very much out of scope for us. I wish you all luck in getting what you need from nextcloud.
